tests/test_async_with.py

```diff
@@ -131,7 +131,9 @@ async def test_create_pool_deprecations(mysql_params, loop):
             warnings.simplefilter("always")
```

Comment:
Looking at this test closer, we could probably just suppress all the warnings except for the `DeprecationWarning` instead of trying to guess on what position it'll be under the next Python version.
Ref: https://docs.python.org/3/library/warnings.html#overriding-the-default-filter.

@MateuszCzubak WDYT?

Reply:
@webknjaz I think that makes a lot of sense but it will look weird to limit to `DeprecationWarning` just in this one place and not in the others. Maybe it would be better to switch all related tests like this in a "refactoring commit", before the actual p3.7/p3.8 compatibility commit?

Reply:
Hm actually the are only 2 invokations of `warnings.simplefilter("always")` - I think it should be enough to change them both within the same commit - WDYT?

Reply:
Well, if you want to switch all, this needs to go into a separate PR. Let me merge this one and then we can do a follow-up.